engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/content-hosts-bulk-system-purpose-modal.controller.js

```diff
@@ -0,0 +1,102 @@
+    ['$scope', '$uibModalInstance', 'HostBulkAction', 'Organization', 'CurrentOrganization', 'Notification', 'hostIds',
+    function ($scope, $uibModalInstance, HostBulkAction, Organization, CurrentOrganization, Notification, hostIds) {
+
+        $scope.organization = Organization.get({id: CurrentOrganization}, function(org) {
+            return org.system_purposes.addons.length;
+        });
+
+        $scope.purposeAddonsList = function () {
+            if ($scope.organization.system_purposes && $scope.organization.system_purposes.addons) {
+                $scope.defaultOptions = ['No Change', 'None (Clear)'];
+                $scope.addons = $scope.organization.system_purposes.addons.map(function (addon) {
+                    return addon; });
+                $scope.addonList = $scope.defaultOptions.concat($scope.addons);
+                return $scope.addonList;
+            }
+            return [];
+        };
+
+        $scope.defaultUsages = ['No change', 'None (Clear)', 'Production', 'Development/Test', 'Disaster Recovery'];
+        $scope.defaultRoles = ['No change', 'None (Clear)', 'Red Hat Enterprise Linux Server', 'Red Hat Enterprise Linux Workstation', 'Red Hat Enterprise Linux Compute Node'];
+        $scope.defaultServiceLevels = ['No change', 'None (Clear)', 'Self-Support', 'Standard', 'Premium'];
+
+        $scope.hostCount = hostIds.included.ids.length;
+
+        $scope.selectedUsages = $scope.defaultUsages[0];
+        $scope.selectedRoles = $scope.defaultRoles[0];
+        $scope.selectedServiceLevels = $scope.defaultServiceLevels[0];
+
+        $scope.selected = {
+           addons: undefined
+        };
+
+        $scope.selectedItemToParam = function (item) {
+            var mapping = {
+                "None (Clear)": "",
+                "No change": null
+            };
+            if (mapping.hasOwnProperty(item)) {
+                return mapping[item];
+            }
+            return item;
+        };
+
+        function actionParams() {
+            var params = hostIds;
+
+            params['purpose_usage'] = $scope.selectedItemToParam($scope.selectedUsages);
+            params['purpose_role'] = $scope.selectedItemToParam($scope.selectedRoles);
+            params['purpose_addons'] = $scope.selectedItemToParam($scope.selectedAddons);
+            params['service_level'] = $scope.selectedItemToParam($scope.selectedServiceLevels);
+
+            if (angular.isUndefined(params['purpose_addons'])) {
+                params['purpose_addons'] = null;
+            } else if (params['purpose_addons'].includes('No Change')) {
+                params['purpose_addons'] = null;
+            } else if (params['purpose_addons'] === "") {
+                params['purpose_addons'] = [];
+            }
+            
+            if (params['purpose_addons'].includes('None (Clear)') && params['purpose_addons'].length > 1) {
```

Comment:
Need to add a check here to make sure `params['purpose_addons']` is defined, in case we just set it to `null` above.  I'm getting a console error 
```
TypeError: Cannot read property 'includes' of null
```

Reply:
@jeremylenz updated and not getting the null anymore